### PR TITLE
misc: add pyright ignores for places where it's not going to be happy

### PR DESCRIPTION
**Pull request by superlopuh**
Pyright used to be happy to accept type expressions as types, but is happy no longer. This adds the required pyright ignores to work around this restriction.

Pylance errors: 94 -> 81

### CODE REVIEW

File: tests/test_is_satisfying_hint.py [hunk 1]
```diff
@@ -372,18 +372,18 @@ def test_parametrized_attribute():
...
-    assert not isa("not this string", Literal["string", "another string"])
-
-    assert isa(1, Literal[1])
-    assert isa(1, Literal[1, 2])
-    assert isa(2, Literal[1, 2])
-    assert not isa(3, Literal[1, 2])
-
-    assert not isa(1, Literal["1"])
-    assert not isa("1", Literal[1])
+    assert isa("string", Literal["string"])  # pyright: ignore[reportArgumentType]
```
Comment (math-fehr) [inline tests/test_is_satisfying_hint.py:375]: Is it the error that `Literal` is a `TypeExpr` and not a `type`?

File: xdsl/tools/tblgen_to_py.py [hunk 2]
```diff
@@ -474,7 +474,7 @@ def generate_op(self, tblgen_op: TblgenOp, dialect_name: str):
                     fields[name] = "var_region_def()"
                 case (True, True):
                     fields[name] = 'var_region_def("single_block")'
-                case _:
+                case _:  # pyright: ignore[reportUnnecessaryComparison]
```
Comment (math-fehr) [inline xdsl/tools/tblgen_to_py.py:477]: Wait, should we just remove the default case then? We had this just because pyright wasn't happy, but now it is

**Review by math-fehr — APPROVED** [review]
Nice! I just had one comment in one place

File: xdsl/tools/tblgen_to_py.py (hunk 2 above)
Comment (superlopuh) [reply xdsl/tools/tblgen_to_py.py:477]: Yep I'll remove the default case with the new pyright, the old one still wants it. I guess I could remove it now and ignore the pyright error on the match case directly, I'll do that quickly.

**Pull request merged**